debian/rules

```diff
@@ -1,114 +1,25 @@
 #!/usr/bin/make -f
-# -*- makefile -*-
-# Sample debian/rules that uses debhelper.
-# GNU copyright 1997 to 1999 by Joey Hess.
 
-# Uncomment this to turn on verbose mode.
 #export DH_VERBOSE=1
 
-CFLAGS = -Wall -Wno-format -g
+export DEB_CFLAGS_MAINT_APPEND  = -Wall -Wno-format -Wno-format-security -g -DTAP_AND_HOLD
+export DEB_LDFLAGS_MAINT_APPEND = -Wl,--as-needed
 
-ifneq (,$(findstring noopt,$(DEB_BUILD_OPTIONS)))
-	CFLAGS += -O0
-else
-	CFLAGS += -O2
-endif
-ifeq (,$(findstring nostrip,$(DEB_BUILD_OPTIONS)))
-	INSTALL_PROGRAM += -s
-endif
-ifneq (,$(findstring thumb,$(DEB_BUILD_OPTIONS)))
-        CFLAGS += -mthumb
-endif
 ifeq (,$(findstring nolauncher,$(DEB_BUILD_OPTIONS)))
 	CONFIGURE_OPTIONS += --enable-maemo-launcher
 	USE_MAEMO_LAUNCHER = true
 endif
 
-CFLAGS += -DTAP_AND_HOLD
-
-CXXFLAGS = $(CFLAGS)
-
-config.status:
-	if [ ! -x configure ]; then ./autogen.sh; fi
-	dh_testdir
-	# Add here commands to configure the package.
-	CFLAGS="$(CFLAGS)" CXXFLAGS="$(CXXFLAGS)" ./configure --prefix=/usr --sysconfdir=/etc --mandir=\$${prefix}/share/man --infodir=\$${prefix}/share/info --disable-static $(CONFIGURE_OPTIONS)
-
-
-build: build-stamp
-
-build-stamp:  config.status
-	dh_testdir
-
-	# Add here commands to compile the package.
-	$(MAKE)
+%:
+	dh $@
 
-	touch build-stamp
+override_dh_configure:
+	dh_configure -- $(CONFIGURE_OPTIONS)
 
-clean:
-	dh_testdir
-	dh_testroot
-	rm -f build-stamp 
-
-	# Add here commands to clean up after the build process.
-	-$(MAKE) distclean
-
-	dh_clean 
-
-install: build
-	dh_testdir
-	dh_testroot
-	dh_clean -k 
-	dh_installdirs
-
-        # Add here commands to install the package into debian/hildon-application-manager.
-	$(MAKE) install DESTDIR=$(CURDIR)/debian/hildon-application-manager
+override_dh_install:
+	dh_install
 	for i in alpha beta; do \
 	  echo $$i; \
 	  mkdir -p $(CURDIR)/debian/hildon-application-manager/usr/share/themes/$$i/backgrounds; \
 	  cp share/themes/$$i-background.png $(CURDIR)/debian/hildon-application-manager/usr/share/themes/$$i/backgrounds/applicationmanager-portrait.png; \
 	done
-	rm `find $(CURDIR)/debian/hildon-application-manager -name '*.la'`
-
-
-# Build architecture-independent files here.
-binary-indep: build install
-# We have nothing to do by default.
-
-# Build architecture-dependent files here.
-binary-arch: build install
-	dh_testdir
-	dh_testroot
-#	dh_installchangelogs ChangeLog
-#	dh_installdocs
-#	dh_installexamples
-#	dh_install
-#	dh_installmenu
-#	dh_installdebconf	
-#	dh_installlogrotate
-#	dh_installemacsen
-#	dh_installpam
-#	dh_installmime
-#	dh_installinit -r -- start 52 2 3 4 . stop 20 0 1 5 6 .
-#	dh_installcron
-#	dh_installinfo
-#	dh_installman
-	dh_installxsession -u 'post 40'
```

Comment:
Why is this line abandoned?

Reply:
Sorry, yes, silently. The ham-after-boot program is used after system updates. It's not something that we do like Fremantle did, and should figure out if and how we want to use ham-after-boot.

I didn't think it was necessary, as it would be obvious that the xsession file is not installed.

Reply:
It would be good, if such things won't be messed with "Port to debhelper 10" commit.
It's a separate action which should be explained in a separate commit message.

Reply:
I still don;t get it why we don;t do it like Fremantle did, care to elaborate?

Reply:
Reverted in 5ae8c60060479faf6cd4c8569988960956d279bb 
You're right :)